sound/soc/sof/topology.c

```diff
@@ -1770,8 +1789,17 @@ static int sof_widget_load_pipeline(struct snd_soc_component *scomp, int index,
 
 	/* send ipc's to create pipeline comp and power up schedule core */
 	ret = sof_load_pipeline_ipc(scomp->dev, pipeline, r);
-	if (ret >= 0)
-		return ret;
+	if (ret < 0)
+		goto err;
+
+	/* set pipe_widget for all widgets with same pipeline_id */
+	list_for_each_entry_reverse(comp_swidget, &sdev->widget_list, list)
+		if (comp_swidget->pipeline_id == swidget->pipeline_id) {
+			comp_swidget->pipe_widget = swidget;
```

Comment:
one question: for multiple pipelines in one pcm, a widget will included by two pipelines. I don't know there is any corner case